src/main/scala/com/microsoft/kusto/spark/utils/KeyVaultUtils.scala

```diff
@@ -4,35 +4,88 @@ import java.io.IOException
 
 import com.microsoft.azure.CloudException
 import com.microsoft.azure.keyvault.KeyVaultClient
-import com.microsoft.kusto.spark.datasource.AadApplicationAuthentication
+import com.microsoft.kusto.spark.datasource._
+
+import scala.util.matching.Regex
 
 object KeyVaultUtils {
   val AppId = "kustoAppId"
   val AppKey = "kustoAppKey"
   val AppAuthority = "kustoAppAuthority"
+  val SasUrl = "blobStorageSasUrl"
+  val StorageAccountId = "blobStorageAccountName"
+  val StorageAccountKey = "blobStorageAccountKey"
+  val Container = "blobContainer"
+  var cachedClient: KeyVaultClient = _
 
-  def getAadParamsFromKeyVaultCertAuth: AadApplicationAuthentication ={
-    // not implemented yet
-    throw new UnsupportedOperationException("does not support cert files yet")
+  private def getClient(clientID: String, clientPassword: String): KeyVaultClient ={
+    if(cachedClient == null) {
+      cachedClient = new KeyVaultADALAuthenticator(clientID, clientPassword).getAuthenticatedClient
+    }
+    cachedClient
   }
 
   @throws[CloudException]
```

Comment:
We don't add these elsewhere right?

Reply:
its just how the Scala treats copied java code - should i delete?